### PR TITLE
make some libraries non required

### PR DESCRIPTION
**Pull request by serdyuka**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -89,7 +89,13 @@
         "symfony/http-kernel": "If you want to use HttpRequestVerifier from symfony's bridge",
         "symfony/http-foundation": "If you want to use HttpRequestVerifier or HttpResponse reply from symfony's bridge",
         "symfony/form": "If you want to use forms",
-        "monolog/monolog": "In case you want to use PSR-3 logger"
+        "monolog/monolog": "In case you want to use PSR-3 logger",
+        "ajbdev/authorizenet-php-api": "If you want to use Authorizenet.net",
```
Comment (makasim) [inline composer.json]: `If you want to use Authorizenet.NET install ajbdev/authorizenet-php-api:1.1.* library`

File: src/Payum/Klarna/Checkout/KlarnaCheckoutGatewayFactory.php [hunk 2]
```diff
@@ -19,6 +19,10 @@ class KlarnaCheckoutGatewayFactory extends GatewayFactory
      */
     protected function populateConfig(ArrayObject $config)
     {
+        if (!class_exists('\\Klarna\\Checkout\\Klarna_Checkout_Exception')) {
```
Comment (makasim) [inline src/Payum/Klarna/Checkout/KlarnaCheckoutGatewayFactory.php]: do not use `\\`

File: src/Payum/Klarna/Invoice/KlarnaInvoiceGatewayFactory.php [hunk 3]
```diff
@@ -29,6 +29,10 @@ class KlarnaInvoiceGatewayFactory extends GatewayFactory
      */
     protected function populateConfig(ArrayObject $config)
     {
+        if (!class_exists('KlarnaException')) {
```
Comment (makasim) [inline src/Payum/Klarna/Invoice/KlarnaInvoiceGatewayFactory.php]: not sure exception is a good class to check for.

File: src/Payum/Klarna/Checkout/KlarnaCheckoutGatewayFactory.php (hunk 2 above)
Comment (makasim) [inline src/Payum/Klarna/Checkout/KlarnaCheckoutGatewayFactory.php]: better to look for Order klarna class.

File: src/Payum/Payex/PayexGatewayFactory.php [hunk 4]
```diff
@@ -33,6 +33,10 @@ class PayexGatewayFactory extends GatewayFactory
      */
     protected function populateConfig(ArrayObject $config)
     {
+        if (!class_exists('SoapClient')) {
+            throw new LogicException('You must install "ext-soap" library.');
```
Comment (makasim) [inline src/Payum/Payex/PayexGatewayFactory.php]: it is not library but extension. make it clear

**Comment by serdyuka** [comment]
@makasim done

**Pull request merged**